components/psram.rst

```diff
@@ -0,0 +1,31 @@
+PSRAM
+=====
+
+.. seo::
+    :description: Configuration for the ESP32 PSRAM platform for ESPHome.
+    :image: psram.svg
+
+This component enables and configures PSRAM if/when available on ESP32 modules/boards.
+It is automatically loaded and enabled by components that require it.
+
+PSRAM is not available on the ESP8266.
```

Comment:
Since there are more than just esp32/esp8266 we should probably flip this
```suggestion
PSRAM is only available on the ESP32.
```

Reply:
Ah yes, good call!

Reply:
...should we also add "and variants"?

Reply:
Nah I think its fine